microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManagerForJava.java

```diff
@@ -214,6 +213,53 @@ public Hover hover(MicroProfileJavaHoverParams params, IJDTUtils utils, IProgres
 		// TODO : aggregate the hover
 		return hovers.get(0);
 	}
+	
+	/**
+	 * Returns the hovered element from the given <code>typeRoot</code>
+	 * and <code>offset</code>. Returns null otherwise
+	 * 
+	 * @param typeRoot the typeRoot
+	 * @param offset   the offset representing the hover location
+	 * @return the hovered element from the given <code>typeRoot</code>
+	 * and <code>offset</code>. Returns null otherwise
+	 * @throws JavaModelException 
+	 */
+	private IJavaElement getHoveredElement(ITypeRoot typeRoot, int offset) throws JavaModelException {
+		IJavaElement hoverElement = typeRoot.getElementAt(offset);
+		if (hoverElement == null) {
+			return null;
+		}
+		if (hoverElement.getElementType() == IJavaElement.METHOD) {
+			hoverElement = getHoveredElementFromMethod((IMethod) hoverElement, offset);
+		}
+		return hoverElement;
+	}
+	
+	/**
+	 * Returns the parameter element from the given <code>method</code> that
+	 * contains the given <code>offset</code>.
+	 * 
+	 * Returns the given <code>method</code> if the correct parameter
+	 * element cannot be found
+	 * 
+	 * @param method the method
+	 * @param offset the offset
+	 * @return the parameter element from the given <code>method</code> that
+	 * contains the given <code>offset</code>
+	 * @throws JavaModelException
+	 */
+	private IJavaElement getHoveredElementFromMethod(IMethod method, int offset) throws JavaModelException {

```

Comment:
Please rename to `getHoveredMethodParameter`

Reply:
fixed